solarforecastarbiter/datamodel.py

```diff
@@ -914,25 +914,47 @@ class ForecastObservation(BaseModel):
     forecast: :py:class:`solarforecastarbiter.datamodel.Forecast`
     observation: :py:class:`solarforecastarbiter.datamodel.Observation`
     reference_forecast: :py:class:`solarforecastarbiter.datamodel.Forecast` or None
-    normalization: float
+    normalization: float or None
+        If None, determined by __set_normalization__
     cost_per_unit_error: float
     """  # NOQA
     forecast: Forecast
     observation: Observation
     reference_forecast: Union[Forecast, None] = None
-    # for now, some function applied to observation (e.g. mean per day)
-    # possible in future
-    normalization: float = 1.0
+    # some function applied to observation (e.g. mean per day)
+    # possible in future. maybe add pd.Series like for
+    # ProcessedForecastObservation
+    normalization: Union[float, None] = None
     # cost: Cost
     cost_per_unit_error: float = 0.0
     data_object: Observation = field(init=False)
 
     def __post_init__(self):
+        if self.normalization is None:
+            __set_normalization__(self)
         object.__setattr__(self, 'data_object', self.observation)
         __check_units__(self.forecast, self.data_object)
         __check_interval_compatibility__(self.forecast, self.data_object)
 
 
+def __set_normalization__(self):
+    if self.observation.variable == 'ac_power':
+        norm = self.observation.site.modeling_parameters.ac_capacity
+    elif self.observation.variable == 'dc_power':
+        norm = self.observation.site.modeling_parameters.ac_capacity
+    elif self.observation.units == 'W/m^2':
+        norm = 1000
```

Comment:
@cwhanse do you think this approach to normalization is reasonable for all irradiance variables or should we treat GHI, DNI, DHI, POA differently?

Reply:
I need a bit of context here - is this attribute used in calculating normalized metrics, or for something else?

Reply:
it is only used to calculate the normalized metrics. should I be thinking more broadly?

Reply:
> Should I be thinking more broadly?
I don't think so. 

`norm = 1000` is a unit conversion. I don't see any advantage to converting from W/m2 to suns. Unlike power forecasts, irradiance forecast metrics are already comparable among locations, without normalization (since the units are W/m2).

If the goal is to express metrics as a percentage of some quantity which is site-specific, then I think we need something like annual (or monthly) insolation.

Reply:
Good point. We're not going to be able to support annual or monthly insolation anytime soon though. Until then, should we refuse to calculate normalized metrics for irradiance and other weather variables? The current behavior is described by this example: if a GHI forecast MAE is 10 (implied units of W/m^2), then its NMAE is 1000 (normalization = 1 W/m^2 and convert to %)

Reply:
10 W/m2 converts to 1000%, that's not intuitive at all. To me, the intent should be to normalize the error in the context of what is 'typical' or 'normal' for the time period. Maybe to the clear-sky insolation, but that requires choosing a clear sky model, and living with it's deficiencies.

I'm ok leaving out normalized metrics for irradiance variables, until there's a clear demand for these metrics.

Reply:
To be clear "current" = "beta4". This PR would convert the 10 W/m^2 to 1%. Anything but a scalar or skipping is out of scope for now.